sklearn/inspection/tests/test_ice.py

```diff
@@ -0,0 +1,132 @@
+"""
+Testing for the ICE module.
+"""
+
+import numpy as np
+import pytest
+
+from sklearn.datasets import make_classification, make_regression
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.ensemble import GradientBoostingRegressor
+from sklearn.inspection import individual_conditional_expectation
+from sklearn.inspection._base import _grid_from_X
+from sklearn.linear_model import LinearRegression
+from sklearn.linear_model import LogisticRegression
+from sklearn.linear_model import MultiTaskLasso
+from sklearn.tree import DecisionTreeRegressor
+from sklearn.utils._testing import assert_array_equal
+
+# toy sample
+X = [[-2, -1], [-1, -1], [-1, -2], [1, 1], [1, 2], [2, 1]]
+y = [-1, -1, -1, 1, 1, 1]
+
+
+# (X, y), n_targets  <-- as expected in the output of partial_dep()
+binary_classification_data = (make_classification(n_samples=50,
+                                                  random_state=0), 1)
+multiclass_classification_data = (make_classification(n_samples=50,
+                                                      n_classes=3,
+                                                      n_clusters_per_class=1,
+                                                      random_state=0), 3)
+regression_data = (make_regression(n_samples=50, random_state=0), 1)
+multioutput_regression_data = (make_regression(n_samples=50, n_targets=2,
+                                               random_state=0), 2)
+
+
+def test_grid_from_X():
```

Comment:
What is the benefit of duplicating this test of a private function?

Reply:
Sorry, this was an oversight when moving testing code around. Fixed in https://github.com/scikit-learn/scikit-learn/pull/16164/commits/154004e41588c4fb9615f408dd4a9cee24b280ca